Update LKG docs spliting PhysX Collider into PhysX Primitive Collider and PhysX Mesh Collider. (#61)
o3de/sig-simulation: testing-guidance/workflow-tests/physics/physics-lkg-workflows.md

|--------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | PhysX Colliders collide with expected entities                     | <ol><li>Create an entity for each of the [PhysX Primitive Collider shape(s)](#physx-collider-shapes), PhysX Mesh Collider or [PhysX Shape Collider shape(s)](#physx-shape-collider-shapes) under test.</li> <li>Add a PhysX Primitive Collider, PhysX Mesh Collider or PhysX Shape Collider with their partner [Shape](#PhysX-Shape-Collider-shapes)<li>Set up a scene in which entities with colliders will collide onto each other.</li><li>Run the Simulation in Editor Game Mode or in the Game Launcher.</li></ol> | <ul><li>PhysX Primitive Colliders components are successfully added to entities.</li><li>PhysX Shape Colliders alert users to the need for corresponding shape meshes.</li><li>PhysX Mesh Collider works when a mesh asset is applied.</li><li>Colliders collide as expected when colliding with entities.</li> |

Reviewer: ```suggestion
| PhysX Colliders collide with expected entities                     | <ol><li>Create an entity for each of the [PhysX Primitive Collider shape(s)](#physx-primitive-collider-shapes), PhysX Mesh Collider or [PhysX Shape Collider shape(s)](#physx-shape-collider-shapes) under test.</li> <li>Add a PhysX Primitive Collider, PhysX Mesh Collider or PhysX Shape Collider with their partner [Shape](#PhysX-Shape-Collider-shapes)<li>Set up a scene in which entities with colliders will collide onto each other.</li><li>Run the Simulation in Editor Game Mode or in the Game Launcher.</li></ol> | <ul><li>PhysX Primitive Colliders components are successfully added to entities.</li><li>PhysX Shape Colliders alert users to the need for corresponding shape meshes.</li><li>PhysX Mesh Collider works when a mesh asset is applied.</li><li>Colliders collide as expected when colliding with entities.</li> |
```

| Collision Groups & Layers Collide according to their matched rules | <ol><li>In the PhysX Configuration Tool create a secondary collision layer and secondary collision group.</li><li>Create enough entities with colliders set up a scene with the desired combinations of Groups and Layer collisions under test.</li><li>Run the Simulation in Editor Game Mode or in the Game Launcher.<ol>                                                                                                                                                    | <ul><li>PhysX Configuration tool opens without issue.</li><li>New Layers and Groups can be assigned via the Layers & Collides with Collider properties.</li><li>PhysX Colliders collide with expected entities according to their Layers & Collision Groups.</li></ul>                                             |

> | PhysX Materials can be applied to colliders and impact their collisions  | <ol><li>Create a PhysX Primitive Collider, PhysX Mesh Collider and PhysX Shape Collider entity setup with appropriate Collider componenets for each PhysX Material under test.</li><li>Add the desired PhysX Material(s) under test to the entities created.</li><li>Set up the scene so that the entities with PhysX Materials under test will collide with something.</li><li>Run the Simulation in Editor Game Mode or in the Game Launcher.</li></ol>                                                     | <ul><li>PhysX Colliders can have PhysX Materials applied.</li><li>PhysX Material properties are applied to Colliders and impact their collisions.</li>                                                                                                                                                                   |

Reviewer: ```suggestion
| PhysX Materials can be applied to colliders and impact their collisions  | <ol><li>Create a PhysX Primitive Collider, PhysX Mesh Collider and PhysX Shape Collider entity setup with appropriate Collider components for each PhysX Material under test.</li><li>Add the desired PhysX Material(s) under test to the entities created.</li><li>Set up the scene so that the entities with PhysX Materials under test will collide with something.</li><li>Run the Simulation in Editor Game Mode or in the Game Launcher.</li></ol>                                                     | <ul><li>PhysX Colliders can have PhysX Materials applied.</li><li>PhysX Material properties are applied to Colliders and impact their collisions.</li>                                                                                                                                                                   |
```

> | PhysX colliders can have their offset and contact offset modified and their collision behavior is impacted accordingly | <ol><li>Set an offset and contact offset for PhysX Mesh Collider and PhysX Primitive colliders with all supported [PhysX Primitive Collider shape(s)](#physx-collider-shapes).</li><li>Set up a scene in which created PhysX colliders with modified offsets and contact offsets collide with simulated rigid bodies moved by PhysX forces.</li><li>Set a contact offset and translation offset for for each of [PhysX Shape Collider shape(s)](#physx-shape-collider-shapes) by using flag enabled Translation Offset (translation offset can be enabled by adding _"EnableShapeComponentTranslationOffset": true_ under _"Amazon" -> "Preferences"_ in _editorpreferences.setreg_ file).</li><li>Set up a scene in which created PhysX Shape colliders with modified shape translation offsets and contact offsets collide with simulated rigid bodies moved by PhysX forces.</li></ol> | <ul><li>PhysX Primitive Colliders can have their offsets and contact offsets modified.</li><ul><li>PhysX Mesh Colliders can have their offsets and contact offsets modified.</li><li>PhysX Shape colliders can have their contact offsets and flag enabled shape translation offsets modified.</li><li>PhysX Primitive colliders, PhysX Mesh colliders and PhysX shape colliders with offsets modified as described above collide with PhysX rigid bodies and other PhysX entities during simulations.</li></ul>                                                     

Reviewer: ```suggestion
| PhysX colliders can have their offset and contact offset modified and their collision behavior is impacted accordingly | <ol><li>Set an offset and contact offset for PhysX Mesh Collider and PhysX Primitive colliders with all supported [PhysX Primitive Collider shape(s)](#physx-primitive-collider-shapes).</li><li>Set up a scene in which created PhysX colliders with modified offsets and contact offsets collide with simulated rigid bodies moved by PhysX forces.</li><li>Set a contact offset and translation offset for for each of [PhysX Shape Collider shape(s)](#physx-shape-collider-shapes) by using flag enabled Translation Offset (translation offset can be enabled by adding _"EnableShapeComponentTranslationOffset": true_ under _"Amazon" -> "Preferences"_ in _editorpreferences.setreg_ file).</li><li>Set up a scene in which created PhysX Shape colliders with modified shape translation offsets and contact offsets collide with simulated rigid bodies moved by PhysX forces.</li></ol> | <ul><li>PhysX Primitive Colliders can have their offsets and contact offsets modified.</li><ul><li>PhysX Mesh Colliders can have their offsets and contact offsets modified.</li><li>PhysX Shape colliders can have their contact offsets and flag enabled shape translation offsets modified.</li><li>PhysX Primitive colliders, PhysX Mesh colliders and PhysX shape colliders with offsets modified as described above collide with PhysX rigid bodies and other PhysX entities during simulations.</li></ul>                                                     
```

> | Force Region Shapes                       | <ol><li>Rerun above scenario but swap out default PhysX Primitive Collider component to support other [PhysX Primitive Collider shapes](#physx-collider-shapes), PhysX Mesh Collider or [PhysX Shape Collider](#PhysX-Shape-Collider-shapes) under test.</li><li>Add the desired [PhysX Force Region Force Type(s)](#PhysX-Force-Region-Force-Types) under test.</li><li>Set up the scene so that an entity will collide with the Force Region under test.</li><li>Run the Simulation in Editor Game Mode or in the Game Launcher.</li></ol>                                                                                                                                   | <ul><li>Force Region Can be applied to Entity with desired PhysX Primitive Collider shape, PhysX Mesh Collider or PhysX Shape Collider w/ desired shape.</li><li>Entities that collide with the Force Region will react according to the the expected Force applied.</li></ul>                                                                                                                                                                                                                                                                                                                         |

Reviewer: ```suggestion
| Force Region Shapes                       | <ol><li>Rerun above scenario but swap out default PhysX Primitive Collider component to support other [PhysX Primitive Collider shapes](#physx-primitive-collider-shapes), PhysX Mesh Collider or [PhysX Shape Collider](#PhysX-Shape-Collider-shapes) under test.</li><li>Add the desired [PhysX Force Region Force Type(s)](#PhysX-Force-Region-Force-Types) under test.</li><li>Set up the scene so that an entity will collide with the Force Region under test.</li><li>Run the Simulation in Editor Game Mode or in the Game Launcher.</li></ol>                                                                                                                                   | <ul><li>Force Region Can be applied to Entity with desired PhysX Primitive Collider shape, PhysX Mesh Collider or PhysX Shape Collider w/ desired shape.</li><li>Entities that collide with the Force Region will react according to the the expected Force applied.</li></ul>                                                                                                                                                                                                                                                                                                                         |
```

| Workflow                                    | Requests                                                                                                                                                                                                             | Things to Watch For                                                                                                                                                                                                                                                                                        |
|---------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Changing PhysX Dynamic Rigid Body Properties in Workflows when using *simulated* type | <ol><li>In any of the workflows with a PhysX Dynamic Rigid Body found on this page change the rigid body properties for the scenario under test.</li><li>Create an entity with a PhysX Dynamic Rigid Body component (type *simulated*) and with each of the supported [PhysX Primitive Collider shape(s)](#physx-collider-shapes), PhysX Mesh Collider or [PhysX Shape Collider shape(s)](#physx-shape-collider-shapes), add a mesh to the entity for visibility, position the entity above the terrain/any PhysX collider and change gravity and time steps values in System and Scene Configuration settings of PhysX Configuration (PREVIEW) tool .</li><li>Enable CCD in PhysX Configuration (PREVIEW) tool and set up a scene in which two or more CCD enabled PhysX rigid bodies collide with each other by using Initial linear velocities with values greater than 50.</li><li>Enable/Disable gravity enabled toggle in PhysX Rigid Body.</li><li>Enable/Disable start asleep toggle in PhysX Rigid Body.</li><li>Set sleep threshold for PhysX Rigid body and set up a scene in which PhysX Rigid body with set sleep treshold is impacted by other PhysX Rigid body.</li><li>Run the Simulation in the Editor Game Mode or in the Game Launcher after providing any of the changes mentioned above to PhysX Rigid body entity.</li></ol> | <ul><li>Scene behaves according to the changes made to rigid body.</li><li>Time step and gravity values set in PhysX Configuration (PREVIEW) tool correctly impact behaviors of rigid bodies in a scene.</li><li>CCD enabled Rigid bodies have better hit detection during scenarios involving big velocities.</li><li>Rigid bodies with gravity enabled toggle turned off do not react to gravity set in PhysX Configuration (PREVIEW)</li><li>Rigid bodies with start asleep toggle turned on have to be woken up by a force to start simulating.</li><li>Rigid bodies with set threshold go to sleep when applied forces are lower than set threshold.</li><li>Extra care should be tested around the following Rigid Body Properties:</li><ul><li>Angular & Linear damping</li><li>COM</li><li>Mass</li><li>Angular Velocity</li><li>Compute Intertia</li></ul></ul> |

Reviewer: ```suggestion
| Changing PhysX Dynamic Rigid Body Properties in Workflows when using *simulated* type | <ol><li>In any of the workflows with a PhysX Dynamic Rigid Body found on this page change the rigid body properties for the scenario under test.</li><li>Create an entity with a PhysX Dynamic Rigid Body component (type *simulated*) and with each of the supported [PhysX Primitive Collider shape(s)](#physx-primitive-collider-shapes), PhysX Mesh Collider or [PhysX Shape Collider shape(s)](#physx-shape-collider-shapes), add a mesh to the entity for visibility, position the entity above the terrain/any PhysX collider and change gravity and time steps values in System and Scene Configuration settings of PhysX Configuration (PREVIEW) tool .</li><li>Enable CCD in PhysX Configuration (PREVIEW) tool and set up a scene in which two or more CCD enabled PhysX rigid bodies collide with each other by using Initial linear velocities with values greater than 50.</li><li>Enable/Disable gravity enabled toggle in PhysX Rigid Body.</li><li>Enable/Disable start asleep toggle in PhysX Rigid Body.</li><li>Set sleep threshold for PhysX Rigid body and set up a scene in which PhysX Rigid body with set sleep threshold is impacted by other PhysX Rigid body.</li><li>Run the Simulation in the Editor Game Mode or in the Game Launcher after providing any of the changes mentioned above to PhysX Rigid body entity.</li></ol> | <ul><li>Scene behaves according to the changes made to rigid body.</li><li>Time step and gravity values set in PhysX Configuration (PREVIEW) tool correctly impact behaviors of rigid bodies in a scene.</li><li>CCD enabled Rigid bodies have better hit detection during scenarios involving big velocities.</li><li>Rigid bodies with gravity enabled toggle turned off do not react to gravity set in PhysX Configuration (PREVIEW)</li><li>Rigid bodies with start asleep toggle turned on have to be woken up by a force to start simulating.</li><li>Rigid bodies with set threshold go to sleep when applied forces are lower than set threshold.</li><li>Extra care should be tested around the following Rigid Body Properties:</li><ul><li>Angular & Linear damping</li><li>COM</li><li>Mass</li><li>Angular Velocity</li><li>Compute Inertia</li></ul></ul> |
```

> | Changing PhysX Dynamic Rigid Body Properties in Workflows when using *kinematic* type | <ol><li>Create two entities with a PhysX Dynamic Rigid Body component (type *kinematic*) and with each of the supported [PhysX Primitive Collider shape(s)](#physx-collider-shapes), PhysX Mesh Collider or [PhysX Shape Collider shape(s)](#physx-shape-collider-shapes), add a mesh to each entity for visibility, position the entities above the terrain.</li><li>Create a script using Script Canvas that moves an entity, test using these different nodes:<ul><li>Move Entity</li><li>Set World Transform</li><li>Set Kinematic Target</li></ul></li><li>Assign the script to one of the entities with a Script Canvas component</li><li>Run the Simulation in the Editor Game Mode or in the Game Launcher.</li></ol> | <ul><li>The Rigid body of the entity with no script will remain still. It collides with other *simulated* bodies.</li><li>The Rigid body of the entity with the script will move accordingly. It pushes other *simulated* bodies it collides with.</li></ul> |

Reviewer: ```suggestion
| Changing PhysX Dynamic Rigid Body Properties in Workflows when using *kinematic* type | <ol><li>Create two entities with a PhysX Dynamic Rigid Body component (type *kinematic*) and with each of the supported [PhysX Primitive Collider shape(s)](#physx-primitive-collider-shapes), PhysX Mesh Collider or [PhysX Shape Collider shape(s)](#physx-shape-collider-shapes), add a mesh to each entity for visibility, position the entities above the terrain.</li><li>Create a script using Script Canvas that moves an entity, test using these different nodes:<ul><li>Move Entity</li><li>Set World Transform</li><li>Set Kinematic Target</li></ul></li><li>Assign the script to one of the entities with a Script Canvas component</li><li>Run the Simulation in the Editor Game Mode or in the Game Launcher.</li></ol> | <ul><li>The Rigid body of the entity with no script will remain still. It collides with other *simulated* bodies.</li><li>The Rigid body of the entity with the script will move accordingly. It pushes other *simulated* bodies it collides with.</li></ul> |
```

> | Importing an .fbx asset into engine and generating physxmesh for it    | <ol><li>Import and find a mesh .fbx asset in Asset Browser.<li>Select and open the.fbx asset with Fbx Settings tool.</ul></li><li>Save the asset physxmesh as a Convex, Primitive and Triangle mesh.</li><li>Use the generated PhysX mesh in a PhysX Mesh Collider.</li></ol> | <ul><li>.fbx assets can be imported into project and are successfully processed by the Asset Processor.</li><li>A physxmesh is generated for an asset when its imported into the project.</li><li>Imported asset can be found in Asset Browser and can be opened with Fbx Settings tool.</li><li>Physxmesh of the asset can be saved as Convex, Primitive and Triangle mesh.</li><li>Saved physxmesh can be used in a PhysX Mesh Collider.</li><li>Triangle meshes are the most accurate meshes but can not be used with simulated rigid bodies.</li><li>All of used physxmeshes are able to generate working colliders for PhysX Mesh Collider component.</li></ul>

Reviewer: ```suggestion
| Importing an .fbx asset into engine and generating physxmesh for it    | <ol><li>Import and find a mesh .fbx asset in Asset Browser.<li>Select and open the .fbx asset with Fbx Settings tool.</ul></li><li>Save the asset physxmesh as a Convex, Primitive and Triangle mesh.</li><li>Use the generated PhysX mesh in a PhysX Mesh Collider.</li></ol> | <ul><li>.fbx assets can be imported into project and are successfully processed by the Asset Processor.</li><li>A physxmesh is generated for an asset when its imported into the project.</li><li>Imported asset can be found in Asset Browser and can be opened with Fbx Settings tool.</li><li>Physxmesh of the asset can be saved as Convex, Primitive and Triangle mesh.</li><li>Saved physxmesh can be used in a PhysX Mesh Collider.</li><li>Triangle meshes are the most accurate meshes but can not be used with simulated rigid bodies.</li><li>All of used physxmeshes are able to generate working colliders for PhysX Mesh Collider component.</li></ul>
```
